control/autoware_autonomous_emergency_braking/src/node.cpp

```diff
@@ -442,29 +430,48 @@ bool AEB::checkCollision(MarkerArray & debug_markers)
     return false;
   }
 
-  auto check_collision = [&](
-                           const auto & path, const colorTuple & debug_colors,
-                           const std::string & debug_ns,
-                           pcl::PointCloud<pcl::PointXYZ>::Ptr filtered_objects) {
+  auto merge_expanded_path_polys = [&](const std::vector<Path> & paths) {
+    std::vector<Polygon2d> merged_expanded_path_polygons;
+    for (const auto & path : paths) {
+      generatePathFootprint(
+        path, expand_width_ + path_footprint_extra_margin_, merged_expanded_path_polygons);
+    }
+    return merged_expanded_path_polygons;
+  };
+
+  auto get_objects_on_path = [&](
+                               const auto & path, PointCloud::Ptr points_belonging_to_cluster_hulls,
+                               const colorTuple & debug_colors, const std::string & debug_ns) {
     // Check which points of the cropped point cloud are on the ego path, and get the closest one
     const auto ego_polys = generatePathFootprint(path, expand_width_);
-    auto objects = std::invoke([&]() {
-      std::vector<ObjectData> objects;
-      // Crop out Pointcloud using an extra wide ego path
-      if (use_pointcloud_data_) {
-        const auto expanded_ego_polys =
-          generatePathFootprint(path, expand_width_ + path_footprint_extra_margin_);
-        cropPointCloudWithEgoFootprintPath(expanded_ego_polys, filtered_objects);
-        const auto current_time = obstacle_ros_pointcloud_ptr_->header.stamp;
-        createObjectDataUsingPointCloudClusters(
-          path, ego_polys, current_time, objects, filtered_objects);
-      }
-      if (use_predicted_object_data_) {
-        createObjectDataUsingPredictedObjects(path, ego_polys, objects);
-      }
-      return objects;
-    });
+    std::vector<ObjectData> objects;
+    // Crop out Pointcloud using an extra wide ego path
+    if (use_pointcloud_data_ && !points_belonging_to_cluster_hulls->empty()) {
+      const auto current_time = obstacle_ros_pointcloud_ptr_->header.stamp;
```

Comment:
@danielsanchezaran 
Unrelated to the changes in this PR, but out of curiosity, why is the timestamp obtained in this way?